src/utils/calldata/cairo.ts

```diff
@@ -102,6 +102,7 @@ export function getAbiContractVersion(abi: Abi): ContractVersion {
  */
 export const uint256 = (it: BigNumberish): Uint256 => {
   const bn = BigInt(it);
+  if (bn < 0) throw Error('uint256 must be positive number');
```

Comment:
3 small nitpicks:
1. `bn` could be renamed to `bi` as is down lower in `bnToUint256`
2. `uint256 must be positive number` could be extracted as constant and added here and in tests instead of c/p
3. I would maybe write: `uint256 must be **a** positive number`

Reply:
Could also call this method within `bnToUint256` and wrap the result properties with `toHex` to remove the duplication.

Reply:
1. Let's be specific and call it bigInt.
2.  dep circle

Reply:
To Fix the dep circle I needed to clean up/restructure (hotfix with the parent file was out of sapient option).
Introduced new Structure Cairo data type, made external optional internal mandatory, backward comp.
b64abccf3888eba49460124a4d7aabbdca1592d1

I imagine every Cairo type to be like this.